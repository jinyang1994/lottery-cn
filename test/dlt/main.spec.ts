import { expect } from 'chai'
import LotteryDlt from 'dlt/index'
import { useMockData } from './helpers'

describe('Test dlt main', function () {
  useMockData()

  it('get point', async function () {
    const dlt = new LotteryDlt(['01', '02', '03', '04', '05'], ['01', '02'])

    dlt.config({
      feedback: output => expect(output).to.matchSnapshot()
    })
    await dlt.check()
  })

  it('get history', async function () {
    const dlt = new LotteryDlt(['01', '02', '03', '04', '05'], ['01', '02'])

    dlt.config({
      feedback: output => expect(output).to.matchSnapshot()
    })
    await dlt.check(22040, 22045)
  })
})
