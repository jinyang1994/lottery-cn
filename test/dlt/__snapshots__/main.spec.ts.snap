// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test dlt main get history 1`] = `
Array [
  Object {
    "blue": Array [
      "01",
      "05",
    ],
    "id": 22045,
    "prizes": Object {
      "1+2": 5,
      "2+0": 5,
      "2+1": 5,
      "2+2": 15,
      "3+0": 5,
      "3+1": 15,
      "3+2": 200,
      "4+0": 100,
      "4+1": 300,
      "4+2": 3000,
      "5+0": 10000,
      "5+1": 136728,
      "5+2": 10000000,
    },
    "red": Array [
      "02",
      "03",
      "07",
      "12",
      "20",
    ],
    "reward": 5,
  },
  Object {
    "blue": Array [
      "03",
      "07",
    ],
    "id": 22044,
    "prizes": Object {
      "1+2": 5,
      "2+0": 5,
      "2+1": 5,
      "2+2": 15,
      "3+0": 5,
      "3+1": 15,
      "3+2": 200,
      "4+0": 100,
      "4+1": 300,
      "4+2": 3000,
      "5+0": 10000,
      "5+1": 174317,
      "5+2": 10000000,
    },
    "red": Array [
      "01",
      "02",
      "16",
      "17",
      "20",
    ],
    "reward": 5,
  },
  Object {
    "blue": Array [
      "09",
      "12",
    ],
    "id": 22043,
    "prizes": Object {
      "1+2": 5,
      "2+0": 5,
      "2+1": 5,
      "2+2": 15,
      "3+0": 5,
      "3+1": 15,
      "3+2": 200,
      "4+0": 100,
      "4+1": 300,
      "4+2": 3000,
      "5+0": 10000,
      "5+1": 38045,
      "5+2": 7059578,
    },
    "red": Array [
      "02",
      "16",
      "17",
      "24",
      "30",
    ],
    "reward": 0,
  },
  Object {
    "blue": Array [
      "02",
      "06",
    ],
    "id": 22042,
    "prizes": Object {
      "1+2": 5,
      "2+0": 5,
      "2+1": 5,
      "2+2": 15,
      "3+0": 5,
      "3+1": 15,
      "3+2": 200,
      "4+0": 100,
      "4+1": 300,
      "4+2": 3000,
      "5+0": 10000,
      "5+1": 184764,
      "5+2": 10000000,
    },
    "red": Array [
      "08",
      "10",
      "11",
      "29",
      "32",
    ],
    "reward": 0,
  },
  Object {
    "blue": Array [
      "06",
      "11",
    ],
    "id": 22041,
    "prizes": Object {
      "1+2": 5,
      "2+0": 5,
      "2+1": 5,
      "2+2": 15,
      "3+0": 5,
      "3+1": 15,
      "3+2": 200,
      "4+0": 100,
      "4+1": 300,
      "4+2": 3000,
      "5+0": 10000,
      "5+1": 126495,
      "5+2": 10000000,
    },
    "red": Array [
      "05",
      "15",
      "25",
      "26",
      "33",
    ],
    "reward": 0,
  },
  Object {
    "blue": Array [
      "08",
      "12",
    ],
    "id": 22040,
    "prizes": Object {
      "1+2": 5,
      "2+0": 5,
      "2+1": 5,
      "2+2": 15,
      "3+0": 5,
      "3+1": 15,
      "3+2": 200,
      "4+0": 100,
      "4+1": 300,
      "4+2": 3000,
      "5+0": 10000,
      "5+1": 67198,
      "5+2": 7441714,
    },
    "red": Array [
      "10",
      "19",
      "25",
      "29",
      "33",
    ],
    "reward": 0,
  },
]
`;

exports[`Test dlt main get point 1`] = `
Array [
  Object {
    "blue": Array [
      "04",
      "06",
    ],
    "id": 22055,
    "prizes": Object {
      "1+2": 5,
      "2+0": 5,
      "2+1": 5,
      "2+2": 15,
      "3+0": 5,
      "3+1": 15,
      "3+2": 200,
      "4+0": 100,
      "4+1": 300,
      "4+2": 3000,
      "5+0": 10000,
      "5+1": 282372,
      "5+2": 10000000,
    },
    "red": Array [
      "01",
      "02",
      "12",
      "15",
      "23",
    ],
    "reward": 5,
  },
]
`;
